src/store/modules/swap/ethBancor.ts

```diff
@@ -1851,43 +1850,33 @@ export class EthBancorModule
           })
         ).catch(e => {
           console.warn("Error fetching ROIs", e);
-        }),
-        Promise.all(
-          allPositions.map(async position => {
-            const currentPoolBalances = await this.fetchRelayBalances({
-              poolId: position.poolToken
-            });
+        })
+      ]);
 
-            const [
-              depositedReserve,
-              opposingReserve
-            ] = sortAlongSide(
-              currentPoolBalances.reserves,
-              reserve => reserve.contract,
-              [position.reserveToken]
-            );
-            const rate0 = new BigNumber(position.reserveRateN)
-              .div(position.reserveRateD)
-              .toString();
-
-            const feeAmountWei = calculatePositionFees(
-              position.poolAmount,
-              currentPoolBalances.smartTokenSupplyWei,
-              position.reserveAmount,
-              depositedReserve.weiAmount,
-              opposingReserve.weiAmount,
-              rate0
-            );
+      const poolReserveIds = allPositions.map(position => {
+        return { poolId: position.poolToken, reserveId: position.reserveToken };
+      });
 
-            const shrunk = shrinkToken(feeAmountWei, 18);
+      const uniquePoolReserveIds = uniqWith(poolReserveIds, isEqual);
 
-            return {
-              positionId: position.id,
-              amount: shrunk
-            };
-          })
-        )
-      ]);
+      const fetchedRewards = await Promise.all(
+        uniquePoolReserveIds.map(async item => {
+          let pendingReserveReward = new BigNumber(0);
+          if (highTierPools.some(x => x === item.poolId)) {
```

Comment:
Lets not assume that the highTierPool cases match that of the item.poolId, lets utilise ```compareString``` here.